torchx/lib/torchx/backend.ex

```diff
@@ -296,6 +296,11 @@ defmodule Torchx.Backend do
     |> to_nx(out)
   end
 
+  @impl true
+  def take(out, t, i, axis) do
+    gather(out, t, Nx.Shape.take_fully_qualified_indices(t, i, axis))
```

Comment:
I am very sorry to inform you that the module `Nx.Shape` is private and we should not access it from `Torchx`. I think we will have to copy and paste the functionality here.

Reply:
Is `take` in EXLA implemented as a gather? Is it performing the same logic as in here? Because if it is, we should consider removing `take` from Nx.Backend and moving this logic to `Nx` itself. This way we get the gradient and this function for free.

Reply:
No problems. In EXLA take is also a gather operation, but it doesn't perform exactly the same logic. We have this logic for the grad implementation of take, though. However, we can implement it through gather + reshape in Nx

Reply:
I see. So what I would do for now is to duplicate the logic and ship this.

Reply:
I've duplicated the logic as discussed, and also took the opportunity to move logic from C++ to Elixir